feat: add design token CSS variables

Piyush-Goenka opened the pull request:
## What kind of change does this PR introduce?
Feature — introduces a centralized Design Token System using CSS variables, with token validation applied via pre-commit hooks and CI.

## Issue Number
Fixes #5274

## Summary
This PR introduces a standardized Design Token System to remove hardcoded UI values and enforce consistency across the application.

### Key changes:
- Added CSS custom properties for colors, spacing, typography, and breakpoints
- Imported tokens globally for application-wide usage
- Added a token validation script to prevent non-token usage
- Integrated validation into pre-commit hooks and CI
- Added developer documentation for token usage and guidelines

This improves UI consistency and significantly reduces future maintenance effort.

## Documentation
Added:  
`docs/docs/docs/developer-resources/design-token-system.md`

## Does this PR introduce a breaking change?
No. Existing styles remain functional.

## Tests
- Token validation script tested locally
- Linting and CI checks verified

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Introduced a comprehensive design token system using CSS custom properties for colors, typography, spacing, and responsive breakpoints to ensure consistent styling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes reviewed (changes requested):
1. Please address this too:

- https://github.com/PalisadoesFoundation/talawa-admin/pull/5843#pullrequestreview-3613792467

1.  In this file `.github/workflows/pull-request.yml` you have made changes to sections that don't concern your work

Piyush-Goenka commented:
In the latest commit, I have made the following changes :-
1. Fixed the failing tests by making changes in the `scripts/__mocks__/@pdfme/generator.spec.ts` , `[src/screens/OrganizationVenues/OrganizationVenues.spec.tsx‎` & `vitest.setup.ts` .
2. Updated `docs/docs/docs/getting-started/installation.md` to add the procedure for creating the virtual environment and to validate python checks.

Piyush-Goenka commented:
@coderabbitai
Review my PR and list down the changes that are needed to be made

palisadoes commented:
@Piyush-Goenka Please ensure coderabbit approves your changes

palisadoes commented:
Please fix the conflicting file in this PR afterwards

Piyush-Goenka commented:
> > > @Piyush-Goenka
> > > To speed things up, please submit a minor PR with a small edit on the same issue. When that is merged you won't have to wait on us to run the GitHub actions manually.
> > 
> > 
> > Should I make a another PR just adding the token directory and adding the token files ??
> 
> Just make a minor edit to some markdown and then revert it in this PR

How can I raise the PR as github actions closes the PR?
Could you tell me the exact procedure so that the github actions doesn't closes my small PR.

palisadoes commented:
> > > > @Piyush-Goenka
> > > > To speed things up, please submit a minor PR with a small edit on the same issue. When that is merged you won't have to wait on us to run the GitHub actions manually.
> > > 
> > > 
> > > Should I make a another PR just adding the token directory and adding the token files ??
> > 
> > Just make a minor edit to some markdown and then revert it in this PR
> 
> How can I raise the PR as github actions closes the PR?
> Could you tell me the exact procedure so that the github actions doesn't closes my small PR.

It's done. The expected automations have been activated. You can work independently now.

Please fix the conflicting files and apply the requested changes. Let us know when the PR is ready for review. 

This is an important, long over due feature. Once this is merged a lot of parallel work can be started.

Are you still working on this?

Piyush-Goenka commented:
@palisadoes 
Since I was travelling I was not able to merge , I am doing it right away.

palisadoes reviewed (changes requested):
@Piyush-Goenka 

1. There are too many files submitted for a relatively simple PR. There are a lot of edits for translations.
2. This is what the issue expects:
    1. #5274
  
        <img width="558" height="302" alt="image" src="https://github.com/user-attachments/assets/cb89f424-53f2-4e6a-b3d7-47f2db716886" />

4. Please fix

Piyush-Goenka commented:
@palisadoes 
Initially I have made changes only in these files but because of the failing tests coderabbit suggested the changes and I made the changes to ensure the tests pass .
So, can I now force push to have a smooth flow.

Piyush-Goenka commented:
I am working on the new issue so should I handle this in the next PR or should I raise a new PR for this ?

palisadoes commented:
> I am working on the new issue so should I handle this in the next PR or should I raise a new PR for this ?

1. Raise a new PR using the issue just discovered in that other PR from the other contributor
2. The script is already updated
